src/data/items.h

```diff
@@ -7211,13 +7236,15 @@ const struct Item gItems[] =
         .fieldUseFunc = ItemUseOutOfBattle_CannotUse,
     },
 
-    [ITEM_ENIGMA_BERRY] = // Todo
+    [ITEM_ENIGMA_BERRY] =
     {
-        .name = _("????????"),
-        .itemId = ITEM_NONE,
-        .price = 0,
-        .description = sDummyDesc,
-        .pocket = POCKET_ITEMS,
+        .name = _("Enigma Berry"),
+        .itemId = ITEM_ENIGMA_BERRY,
+        .price = 20,
+        .holdEffect = HOLD_EFFECT_RESIST_BERRY,
```

Comment:
Enigma Berry should have a different effect from type-resistant berries. You could replace the Gracieda hold effect for it

Reply:
Should it though? They're all berries that share the same trigger condition; being hit by a SE move.

That's why I suggested above to make the Enigma Berry use the same Hold Effect, but unlike the other berries, have its effect tied to `TYPE_NONE` rather than any of the Pokémon types.

Reply:
Yes, because there's a difference between reducing the damage of a move and healing a 1/4 HP of the user.

Reply:
'kay, fair enough.